pymeasure/instruments/temptronic/temptronic_base.py

```diff
@@ -0,0 +1,826 @@
+        "ULIM?", "ULIM %g",
+        """upper air temperature limit.
+
+        Set or get the upper air temperature limit.
+        ULIM nnn -- where nnn is 25 to 225 °C.
+        NOTE: ULIM limits the maximum air temperature in both air and
+        DUT control modes. Additionally, an “out of range” error generates
+        if a setpoint exceeds this value.
+        """,
+        validator=truncated_range,
+        values=[25, 225]
+        )
+
+    temperature_limit_air_dut = Instrument.control(
+        "ADMD?", "ADMD %g",
+        """Air to DUT limit.
+
+        Set the air-to-DUT maximum difference.
+        ADMD nnn -- where nnn is 10 - 300 °C in 1 degree increments.
+        """,
+        validator=truncated_range,
+        values=[10,300]
+        )
+
+    temperature_setpoint = Instrument.control(
+        "SETP?", "SETP %g",
+        """Set or get selected setpoint’s temperature
+
+        Set the currently selected setpoint’s temperature.
+        SETP nnn.n -- where nnn.n is –99.9 to 225.0 °C.
+        NOTE: values entered must be with in the ULIM (the upper limit)
+        and LLIM (the lower limit).
+        """,
+        validator=truncated_range,
+        values=[-99.9, 225]
+        )
+
+    temperature_setpoint_window = Instrument.control(
+        "WNDW?", "WNDW %g",
+        """Setpoint’s temperature window
+
+        Set the currently selected setpoint’s temperature window.
+        WNDW n.n -- where n.n is 0.1 - 9.9 °C
+        NOTE: The window is the maximum positive or negative deviation
+        from the temperature setpoint allowable for an “at temperature”
+        condition.
+        """,
+        validator=truncated_range,
+        values=[0.1, 9.9]
+        )
+
+    temperature_soak_time = Instrument.control(
+        "SOAK?", "SOAK %g",
+        """
+        Set the soak time for the currently selected setpoint.
+        SOAK nnnn – where nnnn is 0 – 9999 seconds.
+        """,
+        validator=truncated_range,
+        values=[0.0,9999]
+        )
+
+    temperature = Instrument.measurement(
+        "TEMP?",
+        """ current main temperature.
+
+        Read the main temperature, in 0.1 °C increments.
+        NOTE: This query returns air temperature when in air-control
+        mode, or DUT temperature when in DUT-control mode.
+        A returned value of greater than 400 degrees indicates an invalid
+        temperature reading.
+        """
+        )
+
+    temperature_condition_status_code = Instrument.measurement(
+        "TECR?",
+        """ temperature condition status register.
+
+        bit 7 –- not used
+        bit 6 -- not used
+        bit 5 -- cycling stopped ("stop on fail" signal was received)
+        bit 4 -- end of all cycles
+        bit 3 -- end of one cycle
+        bit 2 -- end of test (test time has elapsed)
+        bit 1 -- not at temperature
+        bit 0 -- at temperature (soak time has elapsed)
+        """,
+        validator=truncated_range,
+        values=[0,255]
```

Comment:
If you use an `Enum.IntFlag` to store these status bits, and return such a value, you could save all the methods below that query all the possibilities, and get a readable status mask back. You can use the `get_process` and  `set_process` kwargs to define the interaction.

Reply:
E.g. (sketched without checking this)
``` 
class TempStatus(IntFlag):
    at_temperature = 1
    not_at_temperature = 2
    # ...
```
and here use ` get_process= lambda v: TempStatus(v),`, then querying the status should get you ` <TempStatus.not_at_temperature: 2>`, and this behaves like an int, prints nicely, and importantly does all the bit mask resolving automatically, so if multiple flags match you can easily see that.

Reply:
Switched to IntFlag and also introduced those get_processes for most relevant properties (like stated in the other thread).